migrations/035.do.commitments.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE commitments (
+  cid TEXT NOT NULL PRIMARY KEY,
+  published_at TIMESTAMPTZ NOT NULL
+);
+
+CREATE INDEX commitments_published_at ON commitments (published_at);
```

Comment:
TODO: Create a foreign key constraint on the measurements table, once we have commitment rows for all cids

Reply:
Interesting. Are you suggesting creating an FK constraint to ensure `commitments.cid` points to existing `measurements.published_as` values? I think that will prevent us from deleting old measurements.

We can add the constraint in the reverse way - `measurements.published_as` must point to an existing `commitments.cid`. To make this work, I'll need to change my JS code to insert the commitment first and update the measurement afterwards.

What value do you see in this FK constraint?

Reply:
In our chat, we decided the FK is unnecessary because we will be deleting measurement records anyways.